### ISSUE TITLE
Bad state: Cannot add event after closing

### ISSUE THREAD

**Issue by kubastick** [l7ssha/nyxx#35]
How to properly dispose bot?
I'm using these code:
```
await _tmpBot.onReady.drain();
await _tmpBot.close();
await _tmpBot.dispose();
```
And after few minutes I'm getting few hundreds of these exceptions

```
Bad state: Cannot add event after closing
#0      _StreamController.add (dart:async/stream_controller.dart:585:24)
#1      CommonWebSocket.add (package:w_transport/src/web_socket/common/web_socket.dart:124:15)
#2      Shard.send (file:///home/kuba/flutter/flutter/.pub-cache/git/nyxx-flutter-892c2603f34df169771ba9bc3273b8d2b7405485/lib/src/nyxx/Shard.dart:115:18)
#3      Shard._heartbeat (file:///home/kuba/flutter/flutter/.pub-cache/git/nyxx-flutter-892c2603f34df169771ba9bc3273b8d2b7405485/lib/src/nyxx/Shard.dart:121:10)
#4      Shard._handleMsg.<anonymous closure> (file:///home/kuba/flutter/flutter/.pub-cache/git/nyxx-flutter-892c2603f34df169771ba9bc3273b8d2b7405485/lib/src/nyxx/Shard.dart:158:31)
#5      _rootRunUnary (dart:async/zone.dart:1132:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#7      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#8      _CustomZone.bindUnaryCallbackGuarded.<anonymous closure> (dart:async/zone.dart:968:26)
#9      _rootRunUnary (dart:async/zone.dart:1136:13)
#10     _CustomZone.runUnary (da
```

**Comment by l7ssha** [comment]
I think shard isn't really closed after calling `dispose()`. It seems like all events hamdlers are closed but shard still is receiving events and tries to propagate them. I'll take a look.

**Comment by l7ssha** [comment]
So I'm actually closing socket in `dispose()`: 

```dart
  @override
  Future<void> dispose() async {
    await this._socket.close(1000);
  }
```

I added `drain()` to and null assigment to `_socket`. Bassically shard should receive any events after calling `close()` on  `_socket`. This night I'll try to test it.

**Comment by l7ssha** [comment]
Fixed in 6fbd1f1
